lib/lotus/helpers/html_helper/empty_html_node.rb

```diff
@@ -45,13 +59,31 @@ def to_s
         # @api private
         def attributes
           return if @attributes.nil?
-          result = [nil]
+          result = ''
 
-          @attributes.each do |name, value|
-            result << %(#{ name }="#{ value }")
+          @attributes.each do |attribute_name, value|
+            if boolean_attribute?(attribute_name)
+              result << boolean_attribute(attribute_name, value)
+            else
+              result << attribute(attribute_name, value)
+            end
           end
 
-          result.join(ATTRIBUTES_SEPARATOR)
+          result
+        end
+
+        def boolean_attribute?(attribute_name)
+          BOOLEAN_ATTRIBUTES.include?(attribute_name.to_s)
+        end
+
+        # Do not render boolean attributes when their value is _false_.
+        def boolean_attribute(attribute_name, value)
+          return '' if !value
```

Comment:
Thanks for the feedback.

We need to detect 3 cases: 

* boolean attribute and value false: ommit
* boolean attribute and value true: render boolean attribute
* normal attribute

So we can not move value comparison to the boolean_attribute? method (we would need to check if it is a boolean attribute with false value again).

What we could do though is moving the "next" inside the main if as statement modifier or as next. Here a sample with statement modifier:

```ruby
if boolean_attribute?(attribute_name)
  result << boolean_attribute(attribute_name, value) if value
else
 ...
end
```
As for the comparison: i'd use ruby _truthiness_ (everything but nil and false is true) instead of an explicit comparison to true. This is common behavior (not sure how lotus treats it).

What do you think.